feat(*): add docs related to zone egresses (#696)
kumahq/kuma-website: docs/docs/dev/deployments/multi-zone.md

* The **global control plane**:
  * Accept connections only from zone control planes.
  * Accept creation and changes to [policies](/policies) that will be applied to the data plane proxies.
  * Send policies down to zone control planes.
  * Send zone ingresses down to zone control plane.

>   * Keep an inventory of all data planes running in all zones (this is only done for observability but is not required for operations).

Reviewer: ```suggestion
  * Keep an inventory of all data plane proxies running in all zones (this is only done for observability but is not required for operations).
```

* The **zone ingress**:
  * Receive XDS configuration from the local zone control plane.
  * Proxy traffic from other zone data plane proxies to local data plane proxies.
* (optional) The **zone egress**:
    * Receive XDS configuration from the local zone control plane.
    * Proxy traffic from local data plane proxies:

>       * to data plane proxies from other zones;

Reviewer 2: Isn't it to zone ingresses in other zones?

Author: It is indeed

Author: addressed

4. Apply the ingress config, passing the IP address of the zone control plane to `cp-address`:

    ```sh
    kuma-dp run \
    --proxy-type=ingress \
    --cp-address=https://<kuma-cp-address>:5678 \
    --dataplane-token-file=/tmp/ingress-token \
    --dataplane-file=ingress-dp.yaml
    ```

(Optional) If you want to deploy zone egress:

5. Generate the zone token with `egress` in scope:

   To register the zone egress with the zone control plane, we need to generate
   a zone token first with appropriate scope first:

    ```sh
    kumactl generate zone-token --zone=<zone-name> --scope egress > /tmp/zoneegress-token
    ```

   You can also generate the token [with the REST API](../security/zoneegress-auth.md).

> 6. Create an `egress` data plane proxy configuration to allow `kuma-cp` services

Reviewer: ```suggestion
6. Create an egress data plane proxy configuration to allow `kuma-cp` services
```
or
```suggestion
6. Create a `ZoneEgress` data plane proxy configuration to allow `kuma-cp` services
```

   to be configured to proxy traffic to other zones or external services through
   zone egress:

    ```sh
    echo "type: ZoneEgress
    name: zoneegress-01
    networking:
      address: 127.0.0.1 # address that is routable within the zone

>       port: 10002 > zoneegress-dataplane.yaml

Reviewer 2: There's a quote missing here
```suggestion
      port: 10002" > zoneegress-dataplane.yaml
```

When a zone control plane connects to the global control plane, the `Zone` resource is created automatically in the global control plane.

The Zone Ingress tab of the web UI also lists zone control planes that you

> deployed with Zone Ingress.

Reviewer: ```suggestion
deployed with zone ingress.
```

Communication will still work but changes will not be reflected on existing data plane proxies.
:::

### Communication between Global and Zone control plane failing

This can happen with misconfiguration or network connectivity issues between control planes.

* Operations inside the zone will happen correctly (data plane proxies can join, leave and all configuration will be updated and sent correctly).
* Policy changes will not be propagated to the zone control plane.

> * Zone ingress, zone egress and data plane changes will not be propagated to the global control plane:

Reviewer: I could imagine:
```suggestion
* `ZoneIngress`, `ZoneEgress` and `Dataplane` changes will not be propagated to the global control plane:
```
because it's the resources that are changing, or:
```suggestion
* Zone ingress, zone egress and data plane proxy changes will not be propagated to the global control plane:
```

    * The global inventory view of the data plane proxies will be outdated (this only impacts observability).
    * Other zones will not see new services registered inside this zone.
    * Other zones will not see services no longer running inside this zone.
    * Other zones will not see changes in number of instances of each service running in the local zone.
* Global control plane will not send changes from other zone ingress to the zone:
    * Local data plane proxies will not see new services registered in other zones.
    * Local data plane proxies will not see services no longer running in other zones.
    * Local data plane proxies will not see changes in number of instances of each service running in other zones.
* Global control plane will not send changes from other zone ingress to the zone.

::: tip
Note that both local and cross-zone application traffic is not impacted by this failure case.
:::

### Communication between 2 zones failing

This can happen if there are network connectivity issues:

> * between control plane and zone ingress from other zone;

Author: addressed

* Between control plane and zone egress (when present).
* Between zone egress (when present) and zone ingress from other zone.

> * Zone egress (when present) is down.

Reviewer 2: Maybe
```suggestion
* All Zone egress instances of a zone (when present) are down.
```

> * All zone ingresses of a zone are down.

Reviewer 2: Same:
```suggestion
* All zone ingress instances of a zone are down.
```
